src/Tes.ApiClients/HttpApiClient.cs

```diff
@@ -218,13 +268,11 @@ private async Task<HttpRequestMessage> CreateGetHttpRequest(Uri requestUrl, bool
             return httpRequest;
         }
 
-        private static async Task<string> ExecuteRequestAndReadResponseBodyAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        protected async Task<string> ReadResponseBodyAsync(HttpResponseMessage response, CancellationToken cancellationToken)
         {
-            var response = await HttpClient.SendAsync(request, cancellationToken);
-
-            response.EnsureSuccessStatusCode();
-
-            return await response.Content.ReadAsStringAsync(cancellationToken);
+            var ctx = new Polly.Context();
+            ctx.SetOnRetryHandler(LogRetryErrorOnRetryHandler());
+            return await cachingRetryHandler.ExecuteWithRetryAsync(response.Content.ReadAsStringAsync, cancellationToken, ctx);
```

Comment:
Is it valid to retry reading from the stream without making the request again? If the goal here is to retry on socket/io exceptions I believe you must retry by making the request again, not just retrying to read the stream. The runner has a pattern to handle this scenario - by wrapping retriable failures in a retriable exception, which causes the request to be retried.

Reply:
In my experience it has been very valid for many transient TCP/IP stream issues, as long as the remote connection has not closed the stream (which is a configuration many servers follow to improve performance and reduce memory consumption).

There are only two places in the entire codebase where retrying the entire request due to a failure in reading the stream will not retry the entire request with my changes, and that would be easy to remedy: `Tes.ApiClients.HttpApiClient.HttpGetRequestWithRetryPolicyAsync` and `Tes.ApiClients.HttpApiClient.HttpGetRequestWithCachingAndRetryPolicyAsync`. I'll update the PR so that behavior isn't changed.

Reply:
If after your changes a failed stream read will trigger a full request retry, I guess we don't need to retry the ReadAsStringAsync. Am I correct?

Reply:
We don't _need_ to, but if the retry succeeds, we avoid pounding the server with yet another request for basically the same information.

Reply:
I'd recommend deleting it. This method might not be idempotent or able to "rewind" the underlying the stream if it fails. Even if it works, in some cases, in others (more likely) it could cause a futile retry loop that affects performance.

Reply:
Done in my next push to the PR's branch